core/src/main/java/org/stellar/anchor/util/NetUtil.java

```diff
@@ -13,12 +14,30 @@
 import okhttp3.Response;
 
 public class NetUtil {
+
   public static String fetch(String url) throws IOException {
-    Request request = OkHttpUtil.buildGetRequest(url);
-    Response response = getCall(request).execute();
 
-    if (response.body() == null) return "";
-    return Objects.requireNonNull(response.body()).string();
+    try {
+      Request request = OkHttpUtil.buildGetRequest(url);
+      Response response = getCall(request).execute();
+      String message =
+          String.format("NetUtil:fetch of %s unsuccessful. response: %s", url, response.toString());
+
+      // Check if response was unsuccessful (ie not status code 2xx) and throw IOException
+      if (!response.isSuccessful()) {
+        debugF(message);
+        throw new IOException(message);
+      }
+
+      if (response.body() == null) {
+        debugF(message);
+        throw new IOException(message);
+      }
```

Comment:
I think if we use Objects.requireNonNull we will throw an NPE instead of being able to throw a IOException which I think is more appropriate in this context?

Reply:
ah i see your comment. yea, the previous code had both checks. i think we still need the check for null and throwing the IOException but no longer need Objects.requireNotNull